Implement sample test using Playwright

hariclerry opened the pull request:
### Tickets
Closes #5 

### What does this PR do?
This PR implements a sample test using Playwright

### What are the changes made?
- Created a playwright folder inside the tests directory
- Added wikipedia.js with playwright test
- Added a README.md file in the playwright folder with instructions on how to run the test.

### To run test
- Clone the repository and checkout to the `issues-05` branch.
- Run `npm ci` to install dependencies
- Run `node tests/playwright/specs/wikipedia.js` from the app root directory

### Screen shots
None

amire80 committed:
Change the autonym of the Altai language

Making it the same as in language-data. This is the main
variety, so there's no need to call it "Southern".

Change-Id: I8c667b55d59b9c40b00e835f5ff1da6dbdf9bdf7

Commit:
Add db group documentation

Add section to database.md outlining simple
usage of database groups and the 3 remaining
supported types.

Bug: T267077
Change-Id: I66cb2fb063f291afc94ff207ef8d9a035bfc6a86

Commit:
Add baseline MW REST API client for frontend

Bug: T268680
Change-Id: Ib09bf6b1e8cbfb3c51cfed24fc2e1e5bfc14a6eb

cscott committed:
ApiQuerySiteInfo: Replace 'titleconversion' with 'linkconversion' boolean

The 'linkconversion' boolean is also false if 'langconversion' is false.

You could enable title conversion but disable language conversion, but
that's really meaningless; there's no reason to expose that detail.
All of the WMF production sites enable both language conversion and
title/link conversion, so this will make no difference on any of our
production wikis.

Leave the 'titleconversion' field around in the siteinfo for backward
compatibility.

Followup-To: Icc4687638ff1815003dd903854efdbd904854f1e
Change-Id: I8959c9feb60af4cf4de1a2d44090f14a0a2f5a4d

umherirrender committed:
Improve code around internal queries in sqlite database class

Use database quote function for the internal table names
Rename variable to avoid reuse and not confuse taint

This makes taint happy

Bug: T216348
Change-Id: Ic35f63857e85cc523d655aad18384d5bfbc48420

reedy committed:
Add some newlines to MySQL patches for readability

Change-Id: Ice8f297484a03edda1c9e48990360e2098d15133

tgr committed:
Filter out invalid globals in shell.php global hack

Also fix $_REQUEST missing from the superglobal list.

Bug: T208304
Change-Id: Id336cdbb78492d2519b5f2dde01976bbbc17557a

Ammarpad committed:
Rename all PG logging indexes

with exception of logging_actor_time which does not exist
in MySQL and thus will be removed in migrating the table to
abstract schema

New names unify the indexes with MySQL/SQLite names

Bug: T164898
Change-Id: I90bec59b86cdd527d77cc6084056ab4267cc5b40

reedy committed:
Split dropping of image.img_user_timestamp

Splits the dropping of image.img_user_timestamp (which was added
in MW 1.29) from patch-drop-image-img_user_timestamp.sql because
if you're upgrading from a much older MW version, the index may
not exist, and will cause the whole
patch-drop-image-img_user_timestamp.sql patch to fail from the
index not existing.

It probably should have been added by the ifTableNotExists/actor
entry with patch-image-user-index-2.sql (as actor wasn't added
until 1.31)...

Also fix up some comments relating to the patch filenames I
missed whilst I split them in T257407.

Bug: T267543
Change-Id: I3cdecf6df459d44b4d51a9ad9e1ff00997ff0c02

Commit:
Make /page/{title}/html emit etags in RESTBase format

RESTBase used to emit ETag in the `"<rev_id>/<render_id>" format.
For the benefit of the clients, preserve the formar.

Render ID is a UUIDv1 uniquely identifying the ParserOutput.
In future it would be used as a stashing key for stash deduplication.
At this time I decided to just attach the render ID as extension data
to our fake ParserOutput. Once we integrate Parsoid more into core,
we will likely move it into a ParserOutput property, or even
replace CacheTime::mCacheTime with a UUIDv1, but it's too early for that.

Bug: T268234
Change-Id: Ie604e9c98021d59eb1a17ca65f227e8f234a45be

nomoa committed:
Special:Search find variant links before showing the create page link

On a multi–writing system wiki it may happen that Special:Search
show the create page link on a "blue link".
Rely on LanguageConverter::findVariantLink() to find such link.

Bug: T260957
Change-Id: Ic95188543a2aa88764882a7789e4c3a99fd5c3cc

amire80 committed:
Change some instances of "save" to "publish"

For consistency with the "Publish" button.

Change-Id: Ie8a3ff95e46ec8a116e1f6603a380c5643148d1e

reedy committed:
Allow composer/semver 3.2.4

Bug: T269068
Change-Id: Ic6c8974759e2d58f961659631b7698063ea2203b
Depends-On: Icf03053b270b0295e587a497d4ec0efc333cdb69

reedy committed:
Drop composer/semver 1.5.2

Bug: T269068
Change-Id: I4c1eebada36e6f0f6a820eef4d39656d0e1e62c1

reedy committed:
Updating psy/psysh (v0.10.4 => v0.10.5)

Change-Id: If8509f4115a5c9a919234c3433043c2df3f8d8be
Depends-On: I8153cf4b5fd38c2c3984087f2d7ffaf543eb1f21

jdlrobson committed:
Deprecate mediawiki.toc.styles

The only known usage is Cargo skin.
Note, that Minerva references the module but only to remove it from
the page. This is removed in 0e61c55ea.

Follow up to db3c8ea16b3

Bug: T252774
Change-Id: I05d79f073f5303b036e47c2a0b37e9bfbaf0d2d1

hariclerry committed:
setup Microsoft Playwright

hariclerry committed:
setup Microsoft Playwright

hariclerry committed:
setup Microsoft Playwright

hariclerry committed:
implement sample test using Playwright

hariclerry committed:
implement sample test using Playwright

hariclerry committed:
implement sample test using Playwright

Commit:
Merge "Rename all PG logging indexes"

Daimona committed:
Clarify error message in Database::assertBuildSubstringParams

Change-Id: Iba115a6c514b49b02c2073ebfc81b6fc2ec117e9

Commit:
Merge "Allow composer/semver 3.2.4"

Commit:
Merge "Filter out invalid globals in shell.php global hack"

Commit:
Merge "Clarify error message in Database::assertBuildSubstringParams"

Commit:
Merge "Drop composer/semver 1.5.2"

Commit:
Article::view - remove the old subtitle from doOutputFromParserCache.

The method is called from two places:
- For new revision ParserCache, when the subtitle doesn't need
to be applied. That call is protected by !oldid check
- For old revision cache, when subtitle is already applied before
checking the cache. So, no need to set it.

Bug: T269727
Change-Id: Ibec845ccfe43ae2cb4bbe0152c932cb74f3cef4b

sovoid commented on tests/playwright/specs/wikipedia.js:
```diff
@@ -0,0 +1,8 @@
+var { chromium } = require("playwright");
+
+(async () => {
```
I wouldnt call this a test. You need to setup a test runner like jest to run the tests. I guess I wasnt explicit. Ill update the issue

sovoid replied:
Write assertion for an element on the page
https://playwright.dev/#version=v1.6.2&path=docs%2Fassertions.md&q=

hariclerry replied:
This has been noted.

translatewiki committed:
Localisation updates from https://translatewiki.net.

Change-Id: I732ee0183877149286a730c7d398569498d28879

Commit:
Merge "Article::view - remove the old subtitle from doOutputFromParserCache."

Commit:
Merge "Change some instances of "save" to "publish""

Commit:
Merge "Change the autonym of the Altai language"

Volker-E committed:
RCFilters: Temporarily fix TagItemWidget remove button size

To be removed again after OOUI v0.41.1 has been rolled out.

Bug: T269477
Change-Id: Ic0b0fe56cd7eeed09ecadc4c3455869393751403

hariclerry committed:
run eslint in Travis CI

zeljkofilipin committed:
Selenium: fix selenium-daily NPM script

b8a2e21 removed hard-coded values for environment variables from
wdio.conf.js. That caused a couple of Jenkins jobs to fail because
MW_SCRIPT_PATH environment variable was not set.

This patch adds the variable to selenium-daily NPM script, executed by
the failing Jenkins jobs. The alternative is to add the variable to the
Jenkins job configuration.

Bug: T269755
Change-Id: I3c65db297b28449f000a8a79978ba777782f5ffa

hariclerry committed:
run eslint in Travis CI

Commit:
Merge "Selenium: fix selenium-daily NPM script"

Commit:
Merge "Updating psy/psysh (v0.10.4 => v0.10.5)"

Commit:
Merge "Split dropping of image.img_user_timestamp"

Commit:
Merge "ApiQuerySiteInfo: Replace 'titleconversion' with 'linkconversion' boolean"

Commit:
Merge "Deprecate mediawiki.toc.styles"

jdlrobson committed:
Convert mediawiki.skinning/content.css to a LESS file

Simple rename.
This is needed in preparation for
Id20d716ce145e0bae37621fd6e218a793b5332ae

Change-Id: Ic7ae922ae6529c1804178fbe8016736600115a0b

cicalese committed:
Moved page/{title}/bare to PageSourceHandler

Bug: T267981
Change-Id: Ie1a5ee9da5d8231bbf7ea2cbb419ab4bcec33c43

jdlrobson committed:
mw-warning class is replaced with warningbox

The styles can be removed as there are no caching implications here
If a skin is loading `legacy` feature or providing its own (e.g
Minerva)

Non-Wikimedia skins loading content but not legacy will lose
styling here, but that will be consistent with other message
boxes in the interface.

Bug: T269797
Change-Id: I5cfa9d83bedd2786d65d835c1f7623b7a61317c4

cscott committed:
docs: Fix JSON for deprecation in Hooks doc

A quoted string here will fail validation in ExtensionJsonValidationTest.

Change-Id: I8d8bbb03f4238090cab13218d45123e885744c9b

DannyS712 committed:
Emit deprecation warnings for internal EditPage fields

Bug: T252907
Change-Id: I9f0388f1a72bd3253210f8fed2c36a26824a8fde

Commit:
Merge "Add some newlines to MySQL patches for readability"

Commit:
Merge "docs: Fix JSON for deprecation in Hooks doc"

Commit:
Merge "Convert mediawiki.skinning/content.css to a LESS file"

amire80 committed:
Fix grammar in previewconflict

Change-Id: Ib2f71f9c6d5f5576b135d843ac30ed10f63c0406

DannyS712 committed:
ImageRedirectConstraint: accept Content objects

More common interface for constraints

Bug: T157658
Change-Id: Ia50c3fb647ed803a69f5e4511047f1ab88e43bf7

DannyS712 committed:
Add EditPage::handleFailedConstraint

Bug: T157658
Change-Id: I8938c08ce1f7abc2f4c5ae3cb52695c084f6ad2f

Commit:
Merge "Emit deprecation warnings for internal EditPage fields"

Commit:
Merge "Moved page/{title}/bare to PageSourceHandler"

DannyS712 committed:
EditPage::internalAttemptSave - run onEditFilter hook earlier

The conversion of text to a content object will be part of the
eventual backend, while the hook running won't be - flip
the order to make eventual migration easier

Bug: T157658
Change-Id: I9b3a8c5380b76b565be339ccf33537545150c352

Commit:
Merge "ImageRedirectConstraint: accept Content objects"

Commit:
Merge "Add EditPage::handleFailedConstraint"

Commit:
Merge "EditPage::internalAttemptSave - run onEditFilter hook earlier"

Commit:
Merge "Fix grammar in previewconflict"

Commit:
Merge "RCFilters: Temporarily fix TagItemWidget remove button size"

Commit:
Merge "Add db group documentation"

tstarling committed:
eval.php: add --ignore-errors option

An option which causes eval.php to not exit on error. For interactive
debugging.

Change-Id: Ic1db8fd0e5fb4a757985ee8146a3f89a351060c7

translatewiki committed:
Localisation updates from https://translatewiki.net.

Change-Id: I3f3a3dd9bfa09a8cfb377fd6726906b5011390ad

Commit:
Merge "Improve code around internal queries in sqlite database class"

hariclerry committed:
run eslint in Travis CI

lucaswerkmeister committed:
Fix prev/next links on Special:WhatLinksHere

When the previous/next texts are proper links (i.e. we’re not at the end
of that range), we don’t want them to be escaped for the message.

Bug: T269830
Change-Id: I2bd214d8df7153c5780332998600251b191b6396

sovoid committed:
Merge pull request #11 from hariclerry/issues-09

Eslint: Run eslint in Travis CI

hariclerry committed:
Merge branch 'issues-05' of https://github.com/hariclerry/mediawiki into issues-05

Commit:
Merge "Add baseline MW REST API client for frontend"

Commit:
Merge "Make /page/{title}/html emit etags in RESTBase format"

hariclerry committed:
add assertion to playwright test

hariclerry committed:
add assertion to playwright test

Commit:
Merge "Special:Search find variant links before showing the create page link"

Commit:
Merge "mw-warning class is replaced with warningbox"

jdlrobson committed:
i18n: The translation for the search within pages will change

From "containing.." to the more descriptive
"search within pages for the term"

Bug: T249299
Change-Id: I50c177c32c5045654b5db8fd941091251025b7be

Volker-E committed:
OOUI: Backport I18799e54ef46232a54d36e86e2b3d08c3ee0a3d5

Bug: T269477
Change-Id: I468cf0bbee9de896e75a9a3e926d31a9f934e204

Commit:
Merge "eval.php: add --ignore-errors option"

hariclerry committed:
add playwrights rule to eslint

hariclerry committed:
add playwrights rule to eslint

hariclerry committed:
Merge remote-tracking branch 'upstream/master'

hariclerry committed:
Merge branch 'issues-05' of https://github.com/hariclerry/mediawiki into issues-05

sovoid commented:
@hariclerry pulls on upstream should be done in separate pull requests. Notice how the commit history of this PR gets messed up making it harder to review. I am closing this PR. Please make another PR. You can directly pull upstream on the master
